lib/travis/build/addons.rb

```diff
@@ -56,6 +56,8 @@ def addon(name, config)
           const = self.class.const_get(name.to_s.camelize)
           const.new(script, sh, data, config) if const && run_addon?(const)
         rescue NameError
+        rescue
+          raise Travis::Build::AddonConfigError.new(name)
```

Comment:
If we do, we probably don't want to rescue the exception at all, since a new exception will hide `e`. I would also want to avoid putting a lot of logic in `Travis::Build::CompilationError`.